### ISSUE TITLE
TraceBack reports incorrect linenumber

### ISSUE THREAD

**Issue by stinos** [micropython/micropython#2856]
It seems impossible to come up with a simple case to reproduce this but reporting this anyway so it's documented.

Here's a screenshot from code which has the issue, with the linenumbers:
![capture](https://cloud.githubusercontent.com/assets/1902997/22787349/3f3037c8-eedc-11e6-827c-a9346a7c097d.PNG)

In line 86 there's an error because `fr4` is not defined but uPy reports an error for line 87:
```
Traceback (most recent call last):
  File "presenterseq.py", line 89, in <module>
  File "presenterseq.py", line 87, in SeqOrder
NameError: name 'fr4' is not defined
```
Weird thing is, if I just delete line 85 the reported linenumber for the error is correct again. Is this something expected, maybe because of long line length or so?

**Comment by dpgeorge** [comment]
Thanks for the report, this is definitely an interesting edge case!  The minimal example that shows the error is:
```python
def f():
    f([1, 2], [1, 2], [1, 2], {1:1, 1:1, 1:1, 1:1, 1:1, 1:1, 1:X})
    return 1
```
It's now fixed in cc2dbdd1fe4a5131a40b2ab73195786451314bf8, see commit message for a description of the problem.  Test added in e08395a35c37fa7f7c0311cc4c7a71537b8b4227.

**Comment by stinos** [comment]
Thanks!
